terminal/bundles/org.eclipse.terminal.control/src/org/eclipse/terminal/internal/textcanvas/TextCanvas.java

```diff
@@ -215,10 +288,80 @@ public void mouseUp(MouseEvent e) {
 			}
 			redraw();
 		});
-		serVerticalBarVisible(true);
+		setVerticalBarVisible(true);
 		setHorizontalBarVisible(false);
 	}
 
+	private static class Range {
+		final Point start;
+		final Point end;
+
+		Range(Point start, Point end) {
+			this.start = start;
+			this.end = end;
+		}
+
+		boolean isValid() {
+			return start != null && end != null;
+		}
+	}
+
+	private Range wordRangeAt(Point pt) {
```

Comment:
Is null really a valid input here? Are there other cases where a Range can not be supplied? If yes you should use


```suggestion
	private Optional<Range> wordRangeAt(Point pt) {
```

Reply:
I tested different scenarios, and you have is not valid input here. From my perspective, we can remove the null check and there is no need to make the method Optional.